sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java

```diff
@@ -1351,6 +1351,38 @@ public void recognizeCustomFormUrlLabeledDataWithSelectionMark(HttpClient httpCl
         }), SELECTION_MARK_PDF);
     }
 
+    /**
+     * Verifies custom form data for an URL using specified pages.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void recognizeCustomFormUrlLabeledDataWithPages(HttpClient httpClient,
+                                                                      FormRecognizerServiceVersion serviceVersion) {
+        client = getFormRecognizerAsyncClient(httpClient, serviceVersion);
+        urlRunner(fileUrl ->
+            beginTrainingLabeledRunner((trainingFilesUrl, useTrainingLabels) -> {
+                SyncPoller<FormRecognizerOperationResult, CustomFormModel> trainingPoller =
+                    getFormTrainingAsyncClient(httpClient, serviceVersion).beginTraining(trainingFilesUrl,
+                        useTrainingLabels, new TrainingOptions().setPollInterval(durationTestMode))
+                        .getSyncPoller();
+                trainingPoller.waitForCompletion();
+
+                SyncPoller<FormRecognizerOperationResult, List<RecognizedForm>> syncPoller =
+                    client.beginRecognizeCustomFormsFromUrl(
+                        trainingPoller.getFinalResult().getModelId(),
+                        fileUrl,
+                        new RecognizeCustomFormsOptions()
+                            .setFieldElementsIncluded(true)
+                            .setPollInterval(durationTestMode)
+                            .setPages(Collections.singletonList("1")))
+                        .getSyncPoller();
+                syncPoller.waitForCompletion();
+                List<RecognizedForm> recognizedForms = syncPoller.getFinalResult();
+                validateRecognizedResult(recognizedForms, true, true);
+                assertEquals(1, recognizedForms.size());
+            }), FORM_JPG);
```

Comment:
is this testing with a JPEG file? should it test with a file that is capable of having multiple pages (pdf/tiff)?

Reply:
Currently, for custom forms since even multipage content is returned in a single document result we do not have an efficient way of testing this. I will keep a note and add it once we support multi page , per page results for custom forms here.